### PR TITLE
FLUID-6308: Support for one-time passwords when publishing

### PR DESCRIPTION
**Pull request by jobara**
Provides support to pass in one-time passwords for npm two-factor authentication

https://issues.fluidproject.org/browse/FLUID-6308

### CODE REVIEW

**Comment by jobara** [comment]
@amb26 could you please review this PR. I have used it to publish a dev release of itself (2.2.0-dev.20180723T201245Z.c97a732.FLUID-6308) while using 2fa for my NPM account.

File: README.md [hunk 1]
```diff
@@ -186,6 +197,39 @@ publish.standard();
...
+        </tr>
+        <tr>
+            <td>
+                <code>otpFlag</code>
+            </td>
+            <td>
+                Applies the otp flag to the CLI to execute a publish command. This is needed if the NPM account uses two-factor authentication and will only be applied if the otp option is set.
+                <ul>
+                    <li>
+                        <code>${command}</code> will be substituted with the command being executed. Either <code>publichCmd</code> or <code>publishDevCmd</code>
```
Comment (amb26) [inline README.md]: typo publich

File: package-lock.json [hunk 2]
```diff
@@ -1,9 +1,24 @@
 {
   "name": "fluid-publish",
-  "version": "2.0.0",
+  "version": "2.1.0",
```
Comment (amb26) [inline package-lock.json]: We should use this opportunity to axe this lockfile - which in any case has a version which doesn't agree with the package.json version

File: publish.js [hunk 3]
```diff
@@ -232,7 +232,7 @@ publish.getDevVersion = function (moduleVersion, options) {
  *
  * @param isTest {Boolean} - indicates if this is a test run or not
  * @param isDev {Boolean} - indicates if this is a development (true) or standard (false) release
- * @param options {Object} - e.g. {"packCmd": "npm pack", "publishCmd": "npm publish", "publishDevCmd": "npm publish --tag", "publishHint": "publish hint", "publishDevHint": "publish dev hint", devTag: "dev"}
+ * @param options {Object} - e.g. {"packCmd": "npm pack", "publishCmd": "npm publish", "publishDevCmd": "npm publish --tag", "publishHint": "publish hint", "publishDevHint": "publish dev hint", devTag: "dev", "otpFlag": "${command}--otp=${otp}", otp=123456}
```
Comment (amb26) [inline publish.js]: Space after {command}? - we should wrap this extremely long line comment

**Comment by jobara** [comment]
@amb26 this PR is ready for another round of review.

File: Gruntfile.js [hunk 4]
```diff
@@ -17,20 +17,18 @@ module.exports = function (grunt) {
...
-            all: ["**/*.js"]
-        },
-        jsonlint: {
-            all: ["*.json", ".*.json", "tests/*.json"]
+        lintAll: {
+            sources: {
+                md: [ "**/*.md", "!node_modules/**/*.md"],
+                js: ["**/*.js"],
+                json: ["*.json", ".*.json", "tests/*.json", "!package-lock.json"],
+                other: ["./.*", "!node_modules/**/*.*", "!./**/.DS_Store"]
```
Comment (amb26) [inline Gruntfile.js]: What files is this rule intended to catch? It seems more reasonable to just list them individually or in groups.

File: Gruntfile.js [hunk 5]
```diff
@@ -17,20 +17,18 @@ module.exports = function (grunt) {
...
         pkg: grunt.file.readJSON("package.json"),
-        eslint: {
-            all: ["**/*.js"]
-        },
-        jsonlint: {
-            all: ["*.json", ".*.json", "tests/*.json"]
+        lintAll: {
+            sources: {
+                md: [ "**/*.md", "!node_modules/**/*.md"],
+                js: ["**/*.js"],
```
Comment (amb26) [inline Gruntfile.js]: Why doesn't this end up hitting js files in node_modules as presumably the negated pattern on line 22 is trying to guard against?

File: Gruntfile.js (hunk 4 above)
Comment (jobara) [reply Gruntfile.js]: Other is for trailing linespaces.
https://www.npmjs.com/package/gpii-grunt-lint-all

File: Gruntfile.js (hunk 5 above)
Comment (jobara) [reply Gruntfile.js]: I'd assume this is because of the .eslintignore file.

File: Gruntfile.js [hunk 6]
```diff
@@ -17,20 +17,18 @@ module.exports = function (grunt) {
...
         // Project package file destination.
         pkg: grunt.file.readJSON("package.json"),
-        eslint: {
-            all: ["**/*.js"]
-        },
-        jsonlint: {
-            all: ["*.json", ".*.json", "tests/*.json"]
+        lintAll: {
+            sources: {
+                md: [ "**/*.md", "!node_modules/**/*.md"],
```
Comment (amb26) [inline Gruntfile.js]: "*.md" seems sufficient here

File: Gruntfile.js (hunk 4 above)
Comment (amb26) [reply Gruntfile.js]: Yes - I expect it makes more sense to just list LICENCE, .gitignore and .eslintignore explicitly than this convoluted pattern. What if someone started to check images, etc. into the repo?

**Pull request merged**